### PR TITLE
[R2] Add AWS-SDK-NET example

### LINKED ISSUE #4683: R2 how to implement in .net project?
### Which Cloudflare product(s) does this pertain to?

R2

### Subject Matter

.Net Example

### Content Location

Hi,
How to use R2 in .net project.
I cant find any document or example anywhere.
Thanks.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by KianNH**
closes #4683 and #6133

### CODE REVIEW

File: content/r2/examples/aws-sdk-net.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+---
+
+{{<render file="_keys.md">}}
+
+This example uses version 3 of the [aws-sdk-net](https://www.nuget.org/packages/AWSSDK.S3) package. You must pass in the R2 configuration credentials when instantiating your `S3` service client:
+
+## Client setup
+
+In this example, we will pass credentials explicitly to the `IAmazonS3` initialisation. If you wish, you can use a
+shared AWS credentials file or the SDK store in-line with other AWS SDKs. Refer to [Configure AWS credentials](https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-creds.html).
```
Comment (deadlypants1973) [inline content/r2/examples/aws-sdk-net.md]:
```suggestion
In this example, we will pass credentials explicitly to the `IAmazonS3` initialisation. If you wish, you can use a shared AWS credentials file or the SDK store in-line with other AWS SDKs. Refer to [Configure AWS credentials](https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-creds.html) for more details.
```

File: content/r2/examples/aws-sdk-net.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	var secretKey = "<SECRET_KEY>";
+	var credentials = new BasicAWSCredentials(accessKey, secretKey);
+	s3Client = new AmazonS3Client(credentials, new AmazonS3Config
+		{
+			ServiceURL = "https://<ACCOUNT_ID>.r2.cloudflarestorage.com",
+		});
+}
+```
+
+## List buckets & objects
```
Comment (deadlypants1973) [inline content/r2/examples/aws-sdk-net.md]:
```suggestion
## List buckets and objects
```

File: content/r2/examples/aws-sdk-net.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+// cat.png
+```
+
+## Upload and retrieve objects
+
+The [PutObjectAsync](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/S3/MIS3PutObjectAsyncPutObjectRequestCancellationToken.html) and [GetObjectAsync](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/S3/MIS3GetObjectAsyncStringStringCancellationToken.html) methods can be used to upload objects and download objects from an R2 bucket respectively.
+
+{{<Aside type="warning">}}
+
+`DisablePayloadSigning = true` must be passed as Cloudflare R2 does not currently support the Streaming SigV4 implementation used by AWSSDK.S3.
```
Comment (deadlypants1973) [inline content/r2/examples/aws-sdk-net.md:75]: is the . in betwen AWSSDK.S3 intentional?

Comment (KianNH) [reply content/r2/examples/aws-sdk-net.md:75]: Yeah - the package name in NuGet is `AWSSDK.S3`

https://www.nuget.org/packages/AWSSDK.S3

File: content/r2/examples/aws-sdk-net.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+layout: example
+---
+
+{{<render file="_keys.md">}}
+
+This example uses version 3 of the [aws-sdk-net](https://www.nuget.org/packages/AWSSDK.S3) package. You must pass in the R2 configuration credentials when instantiating your `S3` service client:
+
+## Client setup
+
+In this example, we will pass credentials explicitly to the `IAmazonS3` initialisation. If you wish, you can use a shared AWS credentials file or the SDK store in-line with other AWS SDKs. Refer to [Configure AWS credentials](https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-creds.html) for more details.
```
Comment (deadlypants1973) [inline content/r2/examples/aws-sdk-net.md]:
```suggestion
In this example, you will pass credentials explicitly to the `IAmazonS3` initialization. If you wish, use a shared AWS credentials file or the SDK store in-line with other AWS SDKs. Refer to [Configure AWS credentials](https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-creds.html) for more details.
```

**Review by deadlypants1973 — APPROVED** [review]

**Pull request merged**